File: microservices/sucursales-service/src/main/java/cl/duoc/levelup/sucursales_service/repository/SucursalRepository.java
package cl.duoc.levelup.sucursales_service.repository;

import cl.duoc.levelup.sucursales_service.model.Sucursal;
import org.springframework.data.jpa.repository.JpaRepository;

public interface SucursalRepository extends JpaRepository<Sucursal, Long> {
}

File: mobile-app/app/src/main/java/com/levelup/data/repository/SucursalRepository.kt
package com.levelup.data.repository

import com.levelup.data.Sucursal
import com.levelup.data.remote.RetrofitClient
import com.levelup.data.remote.dto.SucursalDto

class SucursalRepository {

    private val api = RetrofitClient.sucursalApi

    private fun mapToUi(dto: SucursalDto): Sucursal {
        return Sucursal(
            id = dto.id ?: 0L,
            nombre = dto.nombre,
            direccion = dto.direccion,
            ciudad = dto.ciudad ?: "",
            horario = dto.horario ?: "",
            telefono = dto.telefono ?: ""
        )
    }

    suspend fun obtenerSucursales(): List<Sucursal> {
        return api.getSucursales().map { mapToUi(it) }
    }
}